ttnn/cpp/ttnn/operations/data_movement/pad/device/pad_program_factory.cpp

```diff
@@ -1575,4 +1577,145 @@ operation::ProgramWithCallbacks pad_rm_sharded(
+
+    tt::DataFormat output_cb_data_format = tt::tt_metal::datatype_to_dataformat_converter(output.get_dtype());
+    uint32_t output_shard_cb_index = tt::CBIndex::c_16;
+    tt::tt_metal::CircularBufferConfig output_shard_cb_config =
+        tt::tt_metal::CircularBufferConfig(
+            shard_height_padded * padded_stick_bytes, {{output_shard_cb_index, output_cb_data_format}})
+            .set_page_size(output_shard_cb_index, padded_stick_bytes)
+            .set_globally_allocated_address(*output.buffer());
+    auto output_shard_cb = tt::tt_metal::CreateCircularBuffer(program, total_cores, output_shard_cb_config);
+
+    // construct const buffer with the pad_value
+    tt::DataFormat pad_val_cb_data_format = tt::tt_metal::datatype_to_dataformat_converter(input_tensor.get_dtype());
+    uint32_t pad_val_cb_index = tt::CBIndex::c_1;
+    tt::tt_metal::CircularBufferConfig cb_pad_val_config =
+        tt::tt_metal::CircularBufferConfig(padded_stick_bytes, {{pad_val_cb_index, pad_val_cb_data_format}})
+            .set_page_size(pad_val_cb_index, padded_stick_bytes);
+    auto pad_val_cb = tt::tt_metal::CreateCircularBuffer(program, total_cores, cb_pad_val_config);
+
+    uint32_t W_padding_front_bytes = input_tensor_start[-3] * input_tensor.element_size();
+
+    uint32_t padding_value_as_u32;
+    if (input_tensor.get_dtype() == tt::tt_metal::DataType::BFLOAT16) {
+        uint16_t bfloat_pad_value_bits = bfloat16(pad_value).to_uint16();
+        padding_value_as_u32 = *reinterpret_cast<uint32_t*>(&bfloat_pad_value_bits);
+    } else if (input_tensor.get_dtype() == tt::tt_metal::DataType::FLOAT32) {
+        padding_value_as_u32 = *reinterpret_cast<uint32_t*>(&pad_value);
+    } else {
+        TT_THROW("ttnn.pad: unsupported data type for pad_rm_sharded_stickwise");
+    }
+
+    // FIXME: assumes that this was sharded using DRAM alignment so that gaps are left in the tensor.
+    // if this changes, we should change the stick step to be 16B (L1 alignment).
+    auto dram_alignment_bytes = tt::tt_metal::hal.get_alignment(tt::tt_metal::HalMemType::DRAM);
+    uint32_t padded_stick_step = tt::round_up(
+        padded_stick_bytes, dram_alignment_bytes);  // round padded_stick bytes to a multiple of dram_alignment_bytes
+    uint32_t unpadded_stick_step = tt::round_up(
+        unpadded_stick_bytes,
+        dram_alignment_bytes);  // round unpadded_stick bytes to a multiple of dram_alignment_bytes
+
+    std::vector<uint32_t> reader_ct_args = {
+        (std::uint32_t)unpadded_stick_bytes,
+        (std::uint32_t)padded_stick_bytes,
+        (std::uint32_t)shard_height_unpadded,
+        (std::uint32_t)shard_height_padded,
+        (std::uint32_t)W_padding_front_bytes,
+        (std::uint32_t)input_shard_cb_index,
+        (std::uint32_t)output_shard_cb_index,
+        (std::uint32_t)unpadded_stick_step,
+        (std::uint32_t)padded_stick_step};
```

Comment:
I don't think all of `(std::uint32_t)` are necessary. Can you try to remove them?

Reply:
not a bug in this case, but I keep repeating that we must avoid c-style cast. 
it is a part of cpp core guidelines and there is even a clang tidy check for it.
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/pro-type-cstyle-cast.html

Reply:
Maybe I'll flip that one on for the next sweep.  It has a FIX-IT.

Reply:
Done!